### ISSUE TITLE
coerce from git describe

### ISSUE THREAD

**Issue by xenoterracide** [semver4j/semver4j#216]
git describe has a format

```
[torvalds@g5 git]$ git describe parent
v1.0.4-14-g2414721
```
> i.e. the current head of my "parent" branch is based on v1.0.4, but since it has a few commits on top of that, describe has added the number of additional commits ("14") and an abbreviated object name for the commit itself ("2414721") at the end.

https://git-scm.com/docs/git-describe

I'd love to see this be able to coerced to where the 14-g... is build information. Maybe this is intentionaly prerelease? which is what coerce does now.

**Comment by severn-everett** [comment]
From what I've been able to ascertain [here](https://github.com/semver/semver/issues/394), there's no unified idea of how to reconcile `git describe` versioning with Semver aside from just declaring everything past the patch version as a prerelease. It could be possible to add a flag for parsing this specifically, but making the library opinionated on how exactly to do so might cause issues down the line.

**Comment by xenoterracide** [comment]
right, I honestly kind of assume it's build info. It really doesn't make a lot of sense as being a pre-release, at least not implicitly. Either way it doesn't seem like it splits on the `-`.
